apps/condo/domains/user/schema/UserRightsSet.test.js

```diff
@@ -0,0 +1,509 @@
+                expect(allOrganizations2).toHaveLength(0)
+
+                // Access provided again
+                const [addedAccessUser] = await updateTestUser(support, reader.user.id, {
+                    rightsSet: { connect: { id: rightsSet.id } },
+                })
+                expect(addedAccessUser).toHaveProperty(['rightsSet', 'id'], rightsSet.id)
+                const filteredOrganizations2 = await Organization.getAll(reader, { id: org.id })
+                expect(filteredOrganizations2).toHaveLength(1)
+                expect(filteredOrganizations2).toEqual([expect.objectContaining({
+                    id: org.id,
+                })])
+
+                // User rights set deleted -> User should lose accesses
+                const [deletedSet] = await updateTestUserRightsSet(support, rightsSet.id, {
+                    deletedAt: dayjs().toISOString(),
+                })
+                expect(deletedSet).toHaveProperty('deletedAt')
+                expect(deletedSet.deletedAt).not.toBeNull()
+                const deletedRightsUser = await User.getOne(support, { id: reader.user.id })
+                expect(deletedRightsUser).toHaveProperty('rightsSet', null)
+                const allOrganizations3 = await Organization.getAll(reader, {})
+                expect(allOrganizations3).toHaveLength(0)
+            })
+        })
+        describe('canExecute<Service>', () => {
+            test('Must give access to execute service', async () => {
+                const executor = await makeClientWithNewRegisteredAndLoggedInUser()
+
+                // No access by default
+                await expectToThrowAccessDeniedErrorToResult(async () => {
+                    await registerNewServiceUserByTestClient(executor)
+                })
+
+                // Access provided
+                const [rightsSet] = await createTestUserRightsSet(support, {
+                    canExecuteRegisterNewServiceUser: true,
+                })
+                const [updatedUser] = await updateTestUser(support, executor.user.id, {
+                    rightsSet: { connect: { id: rightsSet.id } },
+                })
+                expect(updatedUser).toHaveProperty(['rightsSet', 'id'], rightsSet.id)
+                const [registeredUser1] = await registerNewServiceUserByTestClient(executor)
+                expect(registeredUser1).toHaveProperty('id')
+
+                // Access removed from specific user
+                const [removedAccessUser] = await updateTestUser(support, executor.user.id, {
+                    rightsSet: { disconnectAll: true },
+                })
+                expect(removedAccessUser).toHaveProperty('rightsSet', null)
+                await expectToThrowAccessDeniedErrorToResult(async () => {
+                    await registerNewServiceUserByTestClient(executor)
+                })
+
+                // Access provided again
```

Comment:
I don't understand what the reason to test case when user has lose his rights and after that you provide another one similar to the first part of the test

Reply:
Because it's 2 cases of loosing access. For example:
1. You create rights set called "Support level 1", and give it to user A and user B, so they both are available to do some stuff.
2. Later, you decided to drop user A from your group, so you're setting their rightsSet to `null` and expect them to loose direct access. And that's what is happening in first part of the test.
3. You can also delete entire rights set, so any users having it should have `null` in their relation (logic you've asked about below) (( P.S. by default keystone do not set null on soft-delete )). And you expect all users to loose relation (even with `flat` utils like `find`/`getByCondition`) and loose access. So that's what is being checked in the second part of the test.